app/src/main/java/org/simple/clinic/bloodsugar/entry/BloodSugarEntryEffectHandler.kt

```diff
@@ -181,4 +184,53 @@ class BloodSugarEntryEffectHandler @AssistedInject constructor(
         .andThen(compareAndUpdateRecordedAt)
         .toSingleDefault(BloodSugarSaved(createNewBloodSugarEntry.wasDateChanged))
   }
+
+  private fun updateBloodSugarEntryTransformer(scheduler: Scheduler): ObservableTransformer<UpdateBloodSugarEntry, BloodSugarEntryEvent> {
+    return ObservableTransformer { updateBloodSugarEntries ->
+      updateBloodSugarEntries
+          .observeOn(scheduler)
+          .map { updateBloodSugarEntry ->
+            val updateMeasurement = getUpdatedBloodSugarMeasurement(updateBloodSugarEntry)
+            updateMeasurement to updateBloodSugarEntry.wasDateChanged
+          }
+          .flatMapSingle { (bloodSugarMeasurement, wasDateChanged) ->
+            storeUpdateBloodSugarMeasurement(bloodSugarMeasurement)
+                .toSingleDefault(BloodSugarSaved(wasDateChanged))
+          }
+          .compose(reportAnalyticsEvents)
+          .cast<BloodSugarEntryEvent>()
+    }
+  }
+
+  private fun getUpdatedBloodSugarMeasurement(updateBloodSugarEntry: UpdateBloodSugarEntry): BloodSugarMeasurement {
+    val bloodSugarMeasurement = getExistingBloodSugarMeasurement(updateBloodSugarEntry.bloodSugarMeasurementUuid)!!
+    val (user, facility) = userAndCurrentFacility().blockingGet()
```

Comment:
`blockingGet` should not be used in production code as it blocks the thread until something is received.
Let's pair on this and see if we can extract this in the Rx chain itself.

Reply:
Yes, but that entire block of code is called on io thread. So it should be good?

Reply:
Yea. But using `blockingGet` is a non-reactive way of handling events. This takes away the advantage of using Rx in the first place. We should extract these values inside the stream itself.